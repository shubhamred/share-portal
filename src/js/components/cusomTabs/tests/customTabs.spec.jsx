import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import CustomTabs from '../customTabs';

configure({ adapter: new Adapter() });

describe('components/Custom Tabs', () => {
  it('Renders default options', () => {
    const component = shallow(
      <CustomTabs />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  describe('Renders with options', () => {
    const props = {
      tabs: ['tab1', 'tab2'],
      activeTab: 'tab1',
      handleTabClick: jest.fn()
    };
    const component = shallow(
      // eslint-disable-next-line react/jsx-props-no-spreading
      <CustomTabs {...props} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
    it('should execute handleClick', () => {
      component.findWhere((node) => node.key() === 'tab2').children().simulate('click');
      expect(props.handleTabClick).toHaveBeenCalledTimes(1);
    });
  });
});
