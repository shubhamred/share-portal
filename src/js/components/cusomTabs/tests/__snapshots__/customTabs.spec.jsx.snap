// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
  xs={12}
>
  <WithStyles(ForwardRef(Grid))
    className="tabContainer"
    container={true}
    direction="row"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="tab1"
    >
      <Button
        disabled={false}
        key="tab1"
        label="tab1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#4754D6",
            "border": "1px solid  #fff",
            "borderRadius": 21,
            "color": "#fff",
            "marginRight": "16px",
            "minWidth": "100px",
          }
        }
        title=""
        type="button"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="tab2"
    >
      <Button
        disabled={false}
        key="tab2"
        label="tab2"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "border": "1px solid  #4754D6",
            "borderRadius": 21,
            "color": "#4754D6",
            "marginRight": "16px",
            "minWidth": "100px",
          }
        }
        title=""
        type="button"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`components/Custom Tabs Renders default options 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
  xs={12}
>
  <WithStyles(ForwardRef(Grid))
    className="tabContainer"
    container={true}
    direction="row"
  />
</WithStyles(ForwardRef(Grid))>
`;
